### PR TITLE
Access control: Add optional hooks

### PR DESCRIPTION
**Pull request by kalleep**
**What this PR does / why we need it**:
Support for hooks is necessary for the team permissions view to be able to add a member to a team when a role is granted to a user for that team.

**Which issue(s) this PR fixes**:
Partially Fixes [#43287](https://github.com/grafana/grafana-enterprise/issues/2517)

**Special notes for your reviewer**:

### CODE REVIEW

**Review by IevaVasiljeva — APPROVED** [review]
Looks good, but I don't have a very good understanding of how we intend to use these hooks, so you might want to wait for a review from someone with more context on this.

**Review by sakjur — APPROVED** [review]
What she said 👆, lgtm

Would be nice to have some tests to see that the functions run correctly 🙂

**Comment by kalleep** [comment]
> Would be nice to have some tests to see that the functions run correctly slightly_smiling_face
@sakjur Added test to cover this

**Review by sakjur — APPROVED** [review]

**Pull request merged**
